config/config.go

```diff
@@ -368,6 +402,10 @@ func (c *configStruct) ChooseToken(endpoint, overridingToken string) string {
 	ep := normalizeEndpoint(endpoint)
 
 	if overridingToken != "" {
+		if _, ok := c.endpoints[ep]; ok {
+			c.endpoints[ep].Token = overridingToken
```

Comment:
Does this need the mutex locking like this?

```go
	c.endpointsMutex.Lock()
	defer c.endpointsMutex.Unlock()
```

Reply:
Since you're checking then modifying, probably so, could have a race condition